### ISSUE TITLE
v2 <Select /> label as a function

### ISSUE THREAD

**Issue by atanasster** [grommet/grommet#1932]
Please consider adding a label function property to custom render the selected item(s) from Select.

Use case scenario: https://grommet-nextjs.herokuapp.com/grommet-multiselect

Source code: https://github.com/atanasster/grommet-nextjs/tree/master/components/grommet/grommet-multiselect

**Comment by atanasster** [comment]
Thanks for the expanded info, however the entered issue was to pass several other props in the addition to the value prop:
`placeholder` and `onChange` - this allows the value/label to be editable (for example a list of tags that can be removed quickly instead of opening the `<Select />` to un-select them). The use-case url has changed to https://grommet-nextjs.herokuapp.com/add-ons/multiselect and then the label prop
```
if (typeof label === 'function') {
    selectLabel = label({ placeholder, value, onChange });
    }
```
If possible to add those two additional props and with this I can get rid of my Select clone, all other functionalities are great, including the virtualization of rendering

**Comment by alansouzati** [comment]
I'm probably missing something. but do you have access to the placeholder and onChange in your end right ? since you passed it to select in the first place? I believe the way value is done today should be enough for your needs. Let me try to create a codesandbox that replicates your multi-select with tags.

**Comment by atanasster** [comment]
Ok, try to replicate, but on the other hand its a non-breaking change that saves duplicating code.

**Comment by alansouzati** [comment]
yeah. after playing with this a bit, I agree with the fact that we need something to allow a special render for `value` (aka selected elements).

I don't think `label` is the right prop. I propose we name it `valueRender`. @ericsoderberghp the problem with `value` as a node is that you need to start taking care of a bunch of things on your own. For example, as value is always defined, the placeholder will never be presented.

Allowing value to stay as it is, we can leverage all the Select code for placeholder, active button for when elements are selected...

I don't think we need any extra argument rather the `currentValue`,

To summarize, this is the api i'm looking at:

```
export default class MySelect extends Component {
  state = { value: ["First", "Third"] };

  render() {
    const { value } = this.state;
    return (
        <Select
          value={value}
          renderValue={currentValue => <Box>{currentValue}</Box>}
          multiple={true}
          onChange={event => this.setState({ value: event.value })}
          options={["First", "Second", "Third"]}
        />
    );
  }
}
```
